src/AppBundle/Command/AmbassadorShiftTimeLogCommand.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// src/AppBundle/Command/AmbassadorShiftTimeLogCommand.php
+namespace AppBundle\Command;
+
+use Swift_Message;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Doctrine\ORM\Query\Expr\Join;
+
+class AmbassadorShiftTimeLogCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+          ->setName('app:shift:send_late_shifters')
+          ->setDescription('Send shifters which are too late')
+          ->setHelp('This command allows you to send alerts for shifters that are too late')
+            ->addOption('emails', null, InputOption::VALUE_OPTIONAL, 'Email recipients (comma separated)')
+            ->addOption('emailTemplate', null, InputOption::VALUE_OPTIONAL, 'Template used in email alerts', 'SHIFT_LATE_ALERT_EMAIL');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $email_template = $input->getOption('emailTemplate');
+
+        $alerts = $this->computeAlerts();
+        $nbAlerts = count($alerts);
+        if ($nbAlerts > 0) {
+            $output->writeln('<fg=cyan;>Found ' . $nbAlerts . ' alerts to send</>');
+            $this->sendAlertsByEmail($input, $output, $alerts, $email_template);
+        } else {
+            $output->writeln('<fg=cyan;>No shift alert to send</>');
+        }
+    }
+
+    private function computeAlerts() {
+        $time_after_which_members_are_late_with_shifts = $this->getContainer()->getParameter('time_after_which_members_are_late_with_shifts');
+
+        $em = $this->getContainer()->get('doctrine')->getManager();
+        $qb = $em->getRepository("AppBundle:Membership")->createQueryBuilder('o');
+        $qb = $qb->leftJoin("o.beneficiaries", "b")
+            ->leftJoin("b.user", "u")
+            ->leftJoin("o.registrations", "r")->addSelect("r");
+        $qb = $qb->andWhere('o.member_number > 0'); //do not include admin user
+        $qb = $qb->leftJoin("o.registrations", "lr", Join::WITH,'lr.date > r.date')->addSelect("lr")
+            ->where('lr.id IS NULL') //registration is the last one registere
```

Comment:
Pas sur de comprendre pourquoi l'`id` doit être NULL ?

Reply:
On fait un left join de la table registration ("r") avec elle-même ("lr") avec comme condition de jointure que la date de la deuxième ("lr") doit être supérieure à la date de la première ("r").

Donc "lr.id IS NULL" identifie la seule ligne qui n’a pas d’autre ligne plus grande qu’elle-même, i.e. la dernière (c’est utilisé comme "last registration" sur la page web ensuite)